pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -1648,6 +1648,15 @@ private void makeStreamConsumer(String reason) {
     _segmentLogger.info("Creating new stream consumer for topic partition {} , reason: {}", _clientId, reason);
     _partitionGroupConsumer =
         _streamConsumerFactory.createPartitionGroupConsumer(_clientId, _partitionGroupConsumptionStatus);
+    try {
+      _partitionGroupConsumer.validateStreamState(_currentOffset);
+      _serverMetrics.setValueOfPartitionGauge(_tableNameWithType, 0,
+              ServerGauge.INVALID_REALTIME_STREAM_STATE_EXCEPTION, 0);
+    } catch (PermanentConsumerException pce) {
```

Comment:
Should we catch this exception here or bubble it up to the next level?

Reply:
I wasnt sure if the ingest should be stopped. After a couple of conversations the feedback is that it should be stopped. I'll bubble up the exception.

Reply:
Going back on this. On more discussion, it is a bad idea to stop ingestion as that will cause more data loss. So the decision right now is to alert but not stop. Also not that the same condition is checked in `RealtimeDataValidationManager` and that increases a meter but does not attempt to stop. So there is a precedent.